src/common/sourceMaps/sourceMapRepository.ts

```diff
@@ -66,14 +66,20 @@ export interface ISearchStrategy {
  * @param fileContents -- Read contents of the file
  */
 export const createMetadataForFile = async (
+  fsUtils: IFsUtils,
   compiledPath: string,
   fileContents?: string,
 ): Promise<Required<ISourceMapMetadata> | undefined> => {
-  if (typeof fileContents === 'undefined') {
-    fileContents = await readfile(compiledPath);
+  let sourceMapUrl;
+  const possibleSourceMapURL = `${compiledPath}.map`;
+  if (await fsUtils.exists(possibleSourceMapURL)) {
```

Comment:
Rather than making a separate `exists` call here, you can pass the siblings elements in the directory to the `fileProcessor` function.

It's called by `readSomething` which is called for the root entry (no siblings) and in `handleDirectoryEntry`. In the latter case, it's called reading either from the cache tree (where `Object.keys(cache.children)` are siblings) or from the filesystem (where `files` are siblings)

Reply:
Updated. I'm not fully sure thou if you meant the same changes